osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/tasks/ScheduledTaskExecutorServiceTest.java

```diff
@@ -102,4 +107,69 @@ public void testRunFunctionalException()
+        new ScheduledTask(MESSAGE_METADATA, DOMAIN, DOMAIN, DATA_OBJECT, SCHEDULED_TIME);
+    scheduledTask.setPending();
+    //    scheduledTask.set
+    scheduledTasks.add(scheduledTask);
+    when(this.scheduledTaskExecutorJobConfig.scheduledTaskPendingDurationMaxSeconds())
+        .thenReturn(0L);
+    when(this.scheduledTaskExecutorJobConfig.scheduledTaskPageSize()).thenReturn(30);
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            any(ScheduledTaskStatusType.class), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(scheduledTasks)
+        .thenReturn(new ArrayList<ScheduledTask>())
+        .thenReturn(new ArrayList<ScheduledTask>());
+
+    this.scheduledTaskExecutorService.processScheduledTasks();
+
+    final ArgumentCaptor<ScheduledTask> scheduledTaskCaptor =
+        ArgumentCaptor.forClass(ScheduledTask.class);
+    verify(this.scheduledTaskRepository).save(scheduledTaskCaptor.capture());
+    final ScheduledTask savedScheduledTask = scheduledTaskCaptor.getValue();
+    assertThat(savedScheduledTask.getStatus()).isEqualTo(ScheduledTaskStatusType.RETRY);
+  }
+
+  @Test
+  public void testRetryStrandedPendingTaskWhenMaxRetrytimeExceeded() {
+
+    // MessageMetadata with expired max retry time
+    final MessageMetadata messageMetadata =
+        new MessageMetadata.Builder()
+            .withDeviceIdentification("deviceId")
+            .withOrganisationIdentification("organisationId")
+            .withCorrelationUid("correlationId")
+            .withMessageType("messageType")
+            .withMessagePriority(4)
+            .withMaxScheduleTime(Instant.now().minus(100, ChronoUnit.SECONDS).toEpochMilli())
+            .build();
+
+    final List<ScheduledTask> scheduledTasks = new ArrayList<>();
+    final ScheduledTask scheduledTask =
+        new ScheduledTask(messageMetadata, DOMAIN, DOMAIN, DATA_OBJECT, SCHEDULED_TIME);
+    scheduledTask.setPending();
+    //    scheduledTask.set
+    scheduledTasks.add(scheduledTask);
+    when(this.scheduledTaskExecutorJobConfig.scheduledTaskPendingDurationMaxSeconds())
+        .thenReturn(0L);
+    when(this.scheduledTaskExecutorJobConfig.scheduledTaskPageSize()).thenReturn(30);
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            any(ScheduledTaskStatusType.class), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(scheduledTasks)
+        .thenReturn(new ArrayList<ScheduledTask>())
+        .thenReturn(new ArrayList<ScheduledTask>());
+
+    this.scheduledTaskExecutorService.processScheduledTasks();
+
+    final ArgumentCaptor<ScheduledTask> scheduledTaskCaptor =
+        ArgumentCaptor.forClass(ScheduledTask.class);
+    verify(this.scheduledTaskRepository).save(scheduledTaskCaptor.capture());
+    final ScheduledTask savedScheduledTask = scheduledTaskCaptor.getValue();
+    assertThat(savedScheduledTask.getStatus()).isEqualTo(ScheduledTaskStatusType.FAILED);
+  }
```

Comment:
The tests only test cases with a single scheduled task. Should we add another test with multiple tasks?

Reply:
Combined the two tests. So now the one test has multiple tasks